pyxnat/core/resources.py

```diff
@@ -134,6 +134,36 @@ def __init__(cls, name, bases, dct):
 
 # generic classes
 
+from . import derivatives
+from .derivatives import freesurfer as fs
+from .derivatives import ashs
+import types
+import pkgutil
+import inspect
+
+def __get_modules__(m):
+    modules = []
+    prefix = m.__name__ + '.'
+    for importer, modname, ispkg in pkgutil.iter_modules(m.__path__, prefix):
+        module = __import__(modname , fromlist='dummy')
+        if not ispkg:
+            modules.append(module)
+        else:
+            modules.extend(__get_modules__(module))
+    return modules
+
+def __find_all_resources__(m):
+    modules = []
+    classes = {}
+    modules = __get_modules__(m)
+    forbidden_classes = []
+    for m in modules:
+        for name, obj in inspect.getmembers(m):
+            if inspect.isfunction(obj) \
+                    and not obj in forbidden_classes:
+                classes.setdefault(m.__name__, []).append(obj)
+    return modules, classes
```

Comment:
I see a bit of mixed terminology in object names (classes, functions and resources) which I believe could be harmonized to improve readability and a better overall understanding of that code part.

Function `__find_all_resources__()` is -to my understanding- getting a list of available *functions* from each module found under `derivatives`. Note that if -for instance- a class with additional functions is found those will *NOT* be gathered. 

Hence some comments/proposals, 
* rename the internal dictionary `classes` to `functions`.
* change the function name (`__find_all_resources__` ) to avoid misunderstandings: XNAT resources (as entities) VERSUS resource' functionalities/actions. Perhaps something like `__find_all_features__` or similar would be more appropriated. Generally speaking having something as follows sounds weird,
```
import pyxnat.core.resources as resources
import pyxnat.core.derivatives as derivatives

resources.__find_all_resources(derivatives)
```

Reply:
You are totally right. This piece of code has actually been recycled from [bbrc-validator](https://gitlab.com/bbrc/xnat/bbrc-validator/) where the collection of `Tests` is based on classes, not on functions like in this case. Apologies for the confusion.

Reply:
Fixed by 7aa89a59f72cf.